app/src/androidTest/java/com/github/onedirection/map/MapFragmentTest.java

```
@@ -156,15 +160,38 @@ public void testAddEventPutsMarkerOnMap() {
         BottomSheetBehavior<View> bsb = getBottomSheetBehavior();
         assertThat(bsb.getState(), is(BottomSheetBehavior.STATE_HIDDEN));
 
-        fragment.getActivity().runOnUiThread(() -> {
+        Semaphore semaphore = new Semaphore(0);
+
+        runOnUiThreadAndWaitEndExecution(() -> {
             // need to zoom to center the marker and make the next click() click it
             mapboxMap.animateCamera(CameraUpdateFactory.newLatLngZoom(pair.second, 15));
+            mapboxMap.addOnCameraIdleListener(semaphore::release);
         });
+        try {
+            semaphore.acquire();
+        } catch (InterruptedException e) {
```

I think this try catch should not exist, or you should rethrow the exception afterwards. Otherwise, an InterruptedException may not be caught and make the test pass even though it should not.

---

True done